### PR TITLE
Add xrefs and some missing example headers

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: stdlib/LinearAlgebra/src/lq.jl [hunk 1]
```diff
@@ -67,15 +67,15 @@ LQPackedQ(factors::AbstractMatrix{T}, τ::Vector{T}) where {T} = LQPackedQ{T,typ
 """
     lq!(A) -> LQ
 
-Compute the LQ factorization of `A`, using the input
+    Compute the [`LQ`](@ref) factorization of `A`, using the input
```
Comment (dkarrasch) [inline stdlib/LinearAlgebra/src/lq.jl]: The 4-spaces shift is probably an accident.

**Review by dkarrasch — COMMENTED** [review]
Just one typo, otherwise LGTM. I'm not sure the xref to `StridedMatrix` is really helpful, and whether it has an entry in the docs in the first place.

File: stdlib/LinearAlgebra/src/lq.jl (hunk 1 above)
Comment (kshyatt) [reply stdlib/LinearAlgebra/src/lq.jl]: Thanks!

**Comment by kshyatt** [comment]
I checked, `StridedMatrix` does have a doc entry but it's pretty short.

**Review by dkarrasch — APPROVED** [review]

**Comment by StefanKarpinski** [comment]
This good to merge? Go for it when you're ready, @kshyatt.

**Pull request merged**